dsa-solutions/lc-solutions/0600-0699/0621-task-scheduler.md

```diff
@@ -0,0 +1,418 @@
+---
+id: task-scheduler
+title: Task Scheduler
+sidebar_label: 0621 - Task Scheduler
+tags:
+  - Heap
+  - Greedy
+  - Hash Table
+description: "This is a solution to the Task Scheduler problem on LeetCode."
+---
+
+## Problem Description
+
+You are given an array of CPU tasks, each represented by letters A to Z, and a cooling time, n. Each cycle or interval allows the completion of one task. Tasks can be completed in any order, but there's a constraint: identical tasks must be separated by at least n intervals due to cooling time.
+
+​Return the minimum number of intervals required to complete all tasks.
+
+### Examples
+
+**Example 1:**
+
+```
+Input: tasks = ["A","A","A","B","B","B"], n = 2
+
+Output: 8
+
+Explanation: A possible sequence is: A -> B -> idle -> A -> B -> idle -> A -> B.
+
+After completing task A, you must wait two cycles before doing A again. The same applies to task B. In the 3rd interval, neither A nor B can be done, so you idle. By the 4th cycle, you can do A again as 2 intervals have passed.
+```
+
+**Example 2:**
+
+```
+Input: tasks = ["A","C","A","B","D","B"], n = 1
+
+Output: 6
+
+Explanation: A possible sequence is: A -> B -> C -> D -> A -> B.
+
+With a cooling interval of 1, you can repeat a task after just one other task.
+```
+
+### Constraints
+
+- $1 <= tasks.length <= 10^4$
```

Comment:
replace `$1 <= tasks.length <= 10^4$` to `$1 \leq \text{tasks.length} \leq 10^4$`